examples/ceed/ex3-volume.h

```diff
@@ -0,0 +1,151 @@
+        // Mass
+        q_data[i + Q * 0] = w[i] * J[i];
+        // Diffusion
+        q_data[i + Q * 1] = w[i] / J[i];
+      }  // End of Quadrature Point Loop
+      break;
+    case 22:
+      CeedPragmaSIMD for (CeedInt i = 0; i < Q; i++) {
+        // J: 0 2   q_data: 1 3   adj(J):  J22 -J12
+        //    1 3           3 2           -J21  J11
+        const CeedScalar J11 = J[i + Q * 0];
+        const CeedScalar J21 = J[i + Q * 1];
+        const CeedScalar J12 = J[i + Q * 2];
+        const CeedScalar J22 = J[i + Q * 3];
+        const CeedScalar qw  = w[i] / (J11 * J22 - J21 * J12);
+
+        // Mass
+        q_data[i + Q * 0] = w[i] * (J11 * J22 - J21 * J12);
+        // Diffusion
+        q_data[i + Q * 1] = qw * (J12 * J12 + J22 * J22);
+        q_data[i + Q * 2] = qw * (J11 * J11 + J21 * J21);
+        q_data[i + Q * 3] = -qw * (J11 * J12 + J21 * J22);
+      }  // End of Quadrature Point Loop
+      break;
+    case 33:
+      CeedPragmaSIMD for (CeedInt i = 0; i < Q; i++) {
+        // Compute the adjoint
+        CeedScalar A[3][3];
+        for (CeedInt j = 0; j < 3; j++) {
+          for (CeedInt k = 0; k < 3; k++) {
+            // Equivalent code with J as a VLA and no mod operations:
+            // A[k][j] = J[j+1][k+1]*J[j+2][k+2] - J[j+1][k+2]*J[j+2][k+1]
+            A[k][j] = J[i + Q * ((j + 1) % 3 + 3 * ((k + 1) % 3))] * J[i + Q * ((j + 2) % 3 + 3 * ((k + 2) % 3))] -
+                      J[i + Q * ((j + 1) % 3 + 3 * ((k + 2) % 3))] * J[i + Q * ((j + 2) % 3 + 3 * ((k + 1) % 3))];
+          }
+        }
+
+        // Compute quadrature weight / det(J)
+        const CeedScalar qw = w[i] / (J[i + Q * 0] * A[0][0] + J[i + Q * 1] * A[0][1] + J[i + Q * 2] * A[0][2]);
+
+        // Mass
+        q_data[i + Q * 0] = w[i] * (J[i + Q * 0] * A[0][0] + J[i + Q * 1] * A[0][1] + J[i + Q * 2] * A[0][2]);
+        // Diffusion
+        // Stored in Voigt convention
+        // 1 6 5
+        // 6 2 4
+        // 5 4 3
+        q_data[i + Q * 1] = qw * (A[0][0] * A[0][0] + A[0][1] * A[0][1] + A[0][2] * A[0][2]);
+        q_data[i + Q * 2] = qw * (A[1][0] * A[1][0] + A[1][1] * A[1][1] + A[1][2] * A[1][2]);
+        q_data[i + Q * 3] = qw * (A[2][0] * A[2][0] + A[2][1] * A[2][1] + A[2][2] * A[2][2]);
+        q_data[i + Q * 4] = qw * (A[1][0] * A[2][0] + A[1][1] * A[2][1] + A[1][2] * A[2][2]);
+        q_data[i + Q * 5] = qw * (A[0][0] * A[2][0] + A[0][1] * A[2][1] + A[0][2] * A[2][2]);
+        q_data[i + Q * 6] = qw * (A[0][0] * A[1][0] + A[0][1] * A[1][1] + A[0][2] * A[1][2]);
+      }  // End of Quadrature Point Loop
+      break;
+  }
+  return CEED_ERROR_SUCCESS;
+}
+
+/// libCEED Q-function for applying a mass + diffusion operator
+CEED_QFUNCTION(apply_mass_diff)(void *ctx, const CeedInt Q, const CeedScalar *const *in, CeedScalar *const *out) {
+  struct BuildContext *build_data = (struct BuildContext *)ctx;
+  // in[0], out[0] have shape [1,   nc=1, Q]
+  // in[1], out[1] have shape [dim, nc=1, Q]
```

Comment:
`nc` is number of components? Should maybe change this to more human name? Assuming this is used in other examples too.

Reply:
For some reason, we didn't used the VLA casts here, but I think it makes the code clearer, so I added it